### PR TITLE
gl: Relax GLSL requirement from 3.3 to 3.1

### PR DESCRIPTION
**Pull request by plaes**
With these changes, it is possible to start dethrace on
Pinebook Pro (Mali-T860 / Panfrost):

```
OpenGL vendor string: Panfrost
OpenGL renderer string: Mali-T860 (Panfrost)
OpenGL core profile version string: 3.1 Mesa 22.3.0-devel (git-d08bd9a8d8)
OpenGL core profile shading language version string: 1.40
```

### CODE REVIEW

**Comment by dethrace-labs** [comment]
thanks! Your core profile is `3.1`, just curious if you set `SDL_GL_CONTEXT_MINOR_VERSION` to `1` without any other changes does that also compile and run?

Just curious, trying to understand the various OpenGL version conventions!

**Comment by plaes** [comment]
> thanks! Your core profile reported in your output is `3.1`, just curious if you tried to set `SDL_GL_CONTEXT_MINOR_VERSION` to `1` without any other changes? Does that also compile and run?
> 
> Just curious, trying to understand the various OpenGL version conventions!



`glFrameBufferTexture` requires at least 3.2.

File: src/harness/io_platforms/sdl_gl.c [hunk 1]
```diff
@@ -149,11 +149,13 @@ tRenderer* Window_Create(char* title, int width, int height, int pRender_width,
         LOG_PANIC("SDL_INIT_VIDEO error: %s", SDL_GetError());
     }
 
+/*
```
Comment (dethrace-labs) [inline src/harness/io_platforms/sdl_gl.c:152]:
```
```suggestion
```
```

Comment (dethrace-labs) [inline src/harness/io_platforms/sdl_gl.c:152]:
```suggestion
```

File: src/harness/io_platforms/sdl_gl.c [hunk 2]
```diff
@@ -149,11 +149,13 @@ tRenderer* Window_Create(char* title, int width, int height, int pRender_width,
...
     if (SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE) != 0) {
         LOG_PANIC("Failed to set SDL_GL_CONTEXT_PROFILE_MASK attribute. %s", SDL_GetError());
     };
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 2);
+*/
```
Comment (dethrace-labs) [inline src/harness/io_platforms/sdl_gl.c:158]:
```suggestion
```

File: src/harness/io_platforms/sdl_gl.c [hunk 3]
```diff
@@ -149,11 +149,13 @@ tRenderer* Window_Create(char* title, int width, int height, int pRender_width,
         LOG_PANIC("SDL_INIT_VIDEO error: %s", SDL_GetError());
     }
 
+/*
     if (SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE) != 0) {
         LOG_PANIC("Failed to set SDL_GL_CONTEXT_PROFILE_MASK attribute. %s", SDL_GetError());
     };
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
     SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 2);
```
Comment (dethrace-labs) [inline src/harness/io_platforms/sdl_gl.c:157]:
```suggestion
    SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 1);
```

File: src/harness/renderers/gl/shaders.c [hunk 4]
```diff
@@ -1,5 +1,6 @@
 
-const char* vs_2d = "#version 330 core\n"
+const char* vs_2d = "#version 130\n"
```
Comment (dethrace-labs) [inline src/harness/renderers/gl/shaders.c:2]:
```suggestion
const char* vs_2d = "#version 140\n"
```

File: src/harness/renderers/gl/shaders.c [hunk 5]
```diff
@@ -10,7 +11,8 @@ const char* vs_2d = "#version 330 core\n"
                     "   TexCoord = aTexCoord;\n"
                     "}\0";
 
-const char* fs_2d = "#version 330 core\n"
+const char* fs_2d = "#version 130\n"
```
Comment (dethrace-labs) [inline src/harness/renderers/gl/shaders.c:14]:
```suggestion
const char* fs_2d = "#version 140\n"
```

File: src/harness/renderers/gl/shaders.c [hunk 6]
```diff
@@ -23,7 +25,8 @@ const char* fs_2d = "#version 330 core\n"
                     "  FragColor = texel;\n"
                     "}\n\0";
 
-const char* vs_3d = "#version 330 core\n"
+const char* vs_3d = "#version 130\n"
```
Comment (dethrace-labs) [inline src/harness/renderers/gl/shaders.c:28]:
```suggestion
const char* vs_3d = "#version 140\n"
```

File: src/harness/renderers/gl/shaders.c [hunk 7]
```diff
@@ -43,7 +46,8 @@ const char* vs_3d = "#version 330 core\n"
                     "   gl_Position = projection * view * vec4(FragPos, 1.0);\n"
                     "}\0";
 
-const char* fs_3d = "#version 330 core\n"
+const char* fs_3d = "#version 130\n"
```
Comment (dethrace-labs) [inline src/harness/renderers/gl/shaders.c:49]:
```suggestion
const char* fs_3d = "#version 140\n"
```

**Review by dethrace-labs — COMMENTED** [review]
I have to keep the core profile active since I'm on a Mac and without the core profile it falls back to OpenGL 2.1.

I suggested some changes to the PR that still work on my machine and _should_ work on yours too, in theory!

The change from `glFrameBufferTexture` to `glFrameBufferTexture2D` is fine, I actually made the same change myself in a different branch

**Comment by dethrace-labs** [comment]
@plaes when you have a minute, could you help validate the suggested changes? thanks!

**Comment by plaes** [comment]
> @plaes when you have a minute, could you help validate the suggested changes? thanks!

Pushed the changes and validated they work too.

**Comment by dethrace-labs** [comment]
thanks very much!

**Comment by dethrace-labs** [comment]
@plaes could you take a quick photo of it running on your pinebook? I'd love to throw it up on our twitter feed :)

**Review by dethrace-labs — APPROVED** [review]

**Pull request merged**